### ISSUE TITLE
Text recognition has problems with ligatures and makes "so d ware" out of "software"

### ISSUE THREAD

**Issue by Calamari** [burtonator/polar-bookshelf#890]
**Describe the bug**
I have a [PDF document](http://erlang.org/download/armstrong_thesis_2003.pdf) where the text recognition while creating annotation produces "so d ware" where it should read "software". I think it has problems deciphering the "ft" when they are bound together through ligatures. It happened more often throughout that PDF.

**To Reproduce**
1. Download above linked PDF
2. go to page 28 (chapter 2.3 Philosophy)
3. Highlight the second sentence ("To make a fault-tolerant software system which behaves reasonably in the presence of so d ware errors we proceed as follows")
4. See differing annotation on the right

**Expected behavior**
It should annotate the sentence like it was highlighted

**Desktop (please complete the following information):**
 - OS: linux
 - App
 - Version 1.17.5

**Comment by burtonator** [comment]
I'll investigate but I think the issue is that the font issues are because it renders the fonts mostly on the canvas and the text is just for copy /paste so is malformed.

We're working on two solutions... the first is to extract the image directly and the second is to work on seeing if we can't get the original text somehow.

**Comment by Calamari** [comment]
Thanks for the answer.

I could also think of a third way to (at least) work around the problem: Make the text that shows up within the annotation sidebar editable, so I can fix up the typos myself. Currently, I have to rewrite the text in a comment again, so I can later on still keep searching for that term in the annotations.

btw: I am not quite sure what is meant with "extracting the image directly", but I hope it doesn't mean, just copying it to the annotations without reading its contents, because that would break the searching for those annotations (if I did get that intention right).

**Comment by burtonator** [comment]
Yes... this is going to be my work around for now - the ability to just edit the text. I want to keep the ORIGINAL test too of course but we're going to let you edit it.  Maybe the next release actually.

**Comment by Calamari** [comment]
That's great, and fast :)
